server/manager/src/main/java/org/apache/accumulo/manager/tableOps/Utils.java

```diff
@@ -59,12 +59,18 @@ public class Utils {
   public static void checkTableDoesNotExist(ServerContext context, String tableName,
       TableId tableId, TableOperation operation) throws AcceptableThriftTableOperationException {
 
-    TableId id = context.getTableNameToIdMap().get(tableName);
-
-    if (id != null && !id.equals(tableId)) {
-      throw new AcceptableThriftTableOperationException(null, tableName, operation,
-          TableOperationExceptionType.EXISTS, null);
+    try {
+      if (context.getZooReader()
+          .exists(context.getZooKeeperRoot() + Constants.ZTABLES + "/" + tableId.canonical())) {
```

Comment:
I think this function needs a better name, should probably be named `checkTableNameDoesNotExists`. Looking at how its used I think it needs to ensure that no table exists with the given name or if it does exists that it has the specified table id.  The changes to this function do not check all tables ids in zookeeper to ensure they do not have the specified name.  The old code was  doing this check, which requires reading all of the table ids in zookeeper.  The function probably needs some javadoc too in addition to a better name.

Reply:
This comment may be relevant https://github.com/apache/accumulo/issues/4684#issuecomment-2184169539

Reply:
I fixed this in ce3eb51. CreateTableIT run time went from 180s to 280s, but still completes with the additional comms to ZooKeeper.